### ISSUE TITLE
CCP should reload on Ctrl-C

### ISSUE THREAD

**Issue by skx** [ivanizag/iz-cpm#17]
The documentation for Function 10: Read Console Buffer says, in the keybinding section:

* CTRL-C	reboots when at the beginning of line

That's something I've been used to pressing at the CCP prompt, because it can trigger the execution of a `$$$.SUB` file (and also allowed you to swap disks back in the day)

However your emulator doesn't reload - while #13 is open you could see the difference in "running" vs "restart" by watching the change of user-number, or drive, and of course until #16 is resolved SUBMIT is a non-issue too.

You catch Ctrl-C for your "exit emulation" so I guess this might need some reworking.    I did the same thing for a long time, but eventually patched in a HALT, EXIT, and QUIT command to the CCP - they just run `HALT` instructions which is enough for my emulator to terminate.

I also defaulted to requiring two CtrlCs in a row to trigger the reboot, but later relented and made this configurable at run-time, via a [bios extension](https://github.com/skx/cpmulator/blob/master/EXTENSIONS.md) and/or CLI flag.

I guess this is a minor issue as reloads happen after control returns from child processes, but I admit I've become very familiar with pressing the key as a matter of habit nowadays.

**Comment by skx** [comment]
FWIW I did change the code to restart, and that was trivial:

```
--- a/src/bios.rs
+++ b/src/bios.rs
@@ -70,14 +70,13 @@ impl Bios {
     }
 
     pub fn stop(&self) -> bool {
-        self.ctrl_c_count > 1
+        self.ctrl_c_count > 0
     }
 
     pub fn execute(&mut self, reg: &mut Registers, call_trace: bool) -> ExecutionResult {
         if self.stop() {
-            // Stop with two control-c
             self.ctrl_c_count = 0;
-            return ExecutionResult::StopConfirm;
+            return ExecutionResult::WarmBoot;
         }
```

The real question isn't so much how to do the reboot, but how you want to handle things.

If you want to continue using Ctrl-C to exit the emulator then the use of of that character is ruled out - if you can use another character, or require "SPACE Ctrl-C Ctrl-C Y" (which might be too wordy to be memorable) then things get easier.

**Comment by ivanizag** [comment]
I was just writing a fix for that. I thought initially that this feature would be on the CCP code, but it is really on BDOS: https://github.com/brouhaha/cpm22/blob/main/bdos.asm#L573

I'll do a fix on BDOS instead of in the BIOS.
